generator/internal/genclient/translator/protobuf/protobuf_test.go

```diff
@@ -0,0 +1,255 @@
+				{Documentation: "A singular field tag = 16", Name: "f_sfixed64", JSONName: "fSfixed64", ID: ".test.Fake.f_sfixed64", Typez: genclient.SFIXED64_TYPE},
+				{Documentation: "A singular field tag = 17", Name: "f_sint32", JSONName: "fSint32", ID: ".test.Fake.f_sint32", Typez: genclient.SINT32_TYPE},
+				{Documentation: "A singular field tag = 18", Name: "f_sint64", JSONName: "fSint64", ID: ".test.Fake.f_sint64", Typez: genclient.SINT64_TYPE},
+			},
+		})
+	} else {
+		t.Errorf("Cannot find message %s in API State", "Fake")
+	}
+}
+
+func TestScalarArray(t *testing.T) {
+	contents := `
+syntax = "proto3";
+package test;
+
+// A test message.
+message Fake {
+    // A repeated field tag = 1
+    repeated double   f_double     = 1;
+		
+	// A repeated field tag = 3
+    repeated int64    f_int64      = 3;
+	
+	// A repeated field tag = 9
+    repeated string   f_string     = 9;
+	
+	// A repeated field tag = 12
+    repeated bytes    f_bytes      = 12;
+}
+`
+	translator := NewTranslator(&Options{
+		Request:     newCodeGeneratorRequest("resources.proto", contents, t),
+		Language:    "not used",
+		OutDir:      "not used",
+		TemplateDir: "not used",
+	})
+
+	api := translator.makeAPI()
+
+	if message := api.State.MessageByID[".test.Fake"]; message != nil {
+		checkMessage(t, *message, genclient.Message{
+			Name:          "Fake",
+			Documentation: "A test message.",
+			Fields: []*genclient.Field{
+				{Repeated: true, Documentation: "A repeated field tag = 1", Name: "f_double", JSONName: "fDouble", ID: ".test.Fake.f_double", Typez: genclient.DOUBLE_TYPE},
+				{Repeated: true, Documentation: "A repeated field tag = 3", Name: "f_int64", JSONName: "fInt64", ID: ".test.Fake.f_int64", Typez: genclient.INT64_TYPE},
+				{Repeated: true, Documentation: "A repeated field tag = 9", Name: "f_string", JSONName: "fString", ID: ".test.Fake.f_string", Typez: genclient.STRING_TYPE},
+				{Repeated: true, Documentation: "A repeated field tag = 12", Name: "f_bytes", JSONName: "fBytes", ID: ".test.Fake.f_bytes", Typez: genclient.BYTES_TYPE},
+			},
+		})
+	} else {
+		t.Errorf("Cannot find message %s in API State", "Fake")
+	}
+}
+
+func TestScalarOptional(t *testing.T) {
+	contents := `
+syntax = "proto3";
+package test;
+
+// A test message.
+message Fake {
+    // An optional field tag = 1
+    optional double   f_double     = 1;
+		
+	// An optional field tag = 3
+    optional int64    f_int64      = 3;
+	
+	// An optional field tag = 9
+    optional string   f_string     = 9;
+	
+	// An optional field tag = 12
+    optional bytes    f_bytes      = 12;
+}
+`
+	translator := NewTranslator(&Options{
+		Request:     newCodeGeneratorRequest("resources.proto", contents, t),
+		Language:    "not used",
+		OutDir:      "not used",
+		TemplateDir: "not used",
+	})
+
+	api := translator.makeAPI()
+
+	if message := api.State.MessageByID[".test.Fake"]; message != nil {
```

Comment:
I think that this will panic if message is nil. Instead do:

```
message, ok := api.StateMessageById[".test.Fake"]
if !ok {
    t.Fatalf("Cannot find message %s in API State", "Fake")
}

checkMessage(...)
```

Reply:
Done.